code/game/objects/items.dm

```diff
@@ -572,3 +574,6 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
 
 /obj/item/proc/is_sharp()
 	return sharpness
+
+/obj/item/proc/can_dismember()
+	return sharpness && w_class >= 3
```

Comment:
Maybe make it return actual probability so you can tweak it per item if needed instead of just binary state of possibility?